templating/base_render_options.go

```diff
@@ -152,6 +159,54 @@ func LoadAndRenderSteps(template *Template, opts *BaseRenderOptions) (string, er
 	return rendered, nil
 }
 
+// ResolveAndMergeSecrets parses the secrets in the template, resolves them using vault providers and adds them to the values map.
+func resolveAndMergeSecrets(ctx context.Context, template *Template, secretResolver *SecretResolver, valueMap *Values) error {
+
+	// Cheap optimization to skip the secrets merging if it doesn't contain "secrets" string in it. Note that the task can
+	// have the string secrets but may not essentially the secrets section.
+	if !strings.Contains(string(template.Data), "secrets") {
```

Comment:
hmm... if we are optimizing for no secrets perhaps we should do the deserialization first and simply check len(deserialized secrets) > 0 so that we don't have to do a full scan here. Of course that will result in more allocations than in the has secrets path. Can we get a better hot/cold path from the CLI parameters?

Reply:
Not sure I understood your comment. Can you please elaborate?